res/controllers/Denon-MC7000-scripts.js

```diff
@@ -189,27 +188,28 @@ MC7000.init = function() {
         engine.makeConnection("[Sampler"+i+"]", "play", MC7000.SamplerLED);
     }
 
-    // Sampler Volume Control
-    MC7000.samplerLevel = function(channel, control, value) {
-        // check if the Sampler Volume is at Zero and if so hide the sampler bank
-        if (value > 0x00) {
-            engine.setValue("[Samplers]", "show_samplers", true);
-        } else {
-            engine.setValue("[Samplers]", "show_samplers", false);
-        }
-        // get the Sampler Rows opened with its details
-        engine.setValue("[SamplerRow1]", "expanded", true);
-        engine.setValue("[SamplerRow2]", "expanded", true);
+    // Activate Timer for Controller Status SysEx to avoid conflicts with Softtakeover
+    engine.beginTimer(3000, MC7000.delayedSysEx, true);
+};
 
-        //control up to 16 sampler volumes with the one knob on the mixer
-        for (var i = 1; i <= 16; i++) {
-            engine.setValue("[Sampler"+i+"]", "pregain", script.absoluteNonLin(value, 0, 1.0, 4.0));
-        }
-    };
+// SysEx message to receive all knob and fader positions
+MC7000.delayedSysEx = function() {
+    var ControllerStatusSysex = [0xF0, 0x00, 0x20, 0x7F, 0x03, 0x01, 0xF7];
+    midi.sendSysexMsg(ControllerStatusSysex, ControllerStatusSysex.length);
+};
 
-    // Activate Softtakeover for Pitch Faders
-    for (i = 1; i <= 4; i++) {
-        engine.softTakeover("[Channel" + i + "]", "rate", true);
+// Sampler Volume Control
+MC7000.samplerLevel = function(channel, control, value) {
```

Comment:
No idea why pre-commit doesn't complain about this...

```suggestion
MC7000.samplerLevel = function(_channel, _control, value) {
```

Reply:
what does _channel, _control, mean with those underlines?

Reply:
Thanks a lot for explanation but both arguments are indeed needed to get that function working. Possibly that is the reason why eslint doesn't complain. So I keep them as they are.

Reply:
> Thanks a lot for explanation but both arguments are indeed needed to get that function working. Possibly that is the reason why eslint doesn't complain. So I keep them as they are.

Where? I function body only uses `value` as far as I can tell.

Reply:
In case they both would not be needed I could delete them but when i do so the sampler volume and show/hide the sampler bank does not work anymore. you possibly can tell even better why. does `script.absoluteNonLin(value, 0, 1.0, 4.0)` needs to get assignment for `channel` and `control`?